boards/shared/Inc/Io/Io_SharedFrequencyOnlyPwmInput.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <stm32f3xx_hal.h>
+#include <stdbool.h>
+
+#include "Io_SharedFrequencyOnlyPwmInputConfig.h"
+
+#ifndef MAX_NUM_OF_FREQ_ONLY_PWM_INPUTS
```

Comment:
I think we should write this `#ifndef` actually because it doesn't scale well. We end up with all the boards having unnecessary `#define`. Let's give it a default value if a board doesn't define `MAX_NUM_OF_FREQ_ONLY_PWM_INPUTS` (and clean up `Io_SharedFrequencyOnlyPwmInputConfig.h` for boards that don't need this module)

```
#ifndef MAX_NUM_OF_FREQ_ONLY_PWM_INPUTS
#define MAX_NUM_OF_FREQ_ONLY_PWM_INPUTS 0
#endif
```

I think a value of `0` might throw a compiler warning, so use `1` if necessary.

Reply:
Fixed